In [1]:
from src.agents.offer_agent.agent import *

Status:  Initialising nodes
Status:  Initialising ChatOpenAI
Status:  ChatOpenAI initialised
Status:  Initialising nodes
Status:  Initialising ChatOpenAI
Status:  ChatOpenAI initialised


In [11]:
segments = []
metrics = ['1', '2', '3', '4', '5', '6', '7']
ideas = ['96']
human_remark = 'Explore personalised in-app offers for my segemnts'

In [12]:
offers = get_offers(metric_ids=metrics, segment_ids=segments, human_remark=human_remark, idea_ids=ideas)

Status:  Plot generated
Status:  Plot generated
Status:  Plot generated
Status:  Plot generated
Status:  Plot generated
Status:  Plot generated
Status:  Plot generated


In [13]:
print(offers['offers'])

```markdown
# Personalised In-App Purchase Bundles Report

## Overview
This report outlines personalised in-app purchase bundle recommendations tailored for specific player segments within the True Survival Series. The focus is on enhancing player engagement through in-game farming mechanics and experiences.

### Bundle 1: Farm Automation Kit Expansion

#### Bundle Price
- **$9.99**

#### Bundle Items
- 5 Advanced Farming Tools (e.g., automated tillers)
- 10 Exclusive Seeds (with special attributes)
- 1 Farm Layout Template (easy to implement)

#### Bundle Description
Boost your automated farming with the **Farm Automation Kit Expansion**! This bundle includes advanced tools designed to enhance productivity, exclusive seeds that yield higher resources, and a blueprint for an efficient farm layout. Perfect for players who wish to maximize their farming potential without requiring design changes or additional development.

#### Segments
- Players engaged primarily in farming activities.


In [14]:
chat_id = register_new_chat(offers)

In [15]:
chat_id

'101'

# Personalised In-App Purchase Bundles Report

## Overview
This report outlines personalised in-app purchase bundle recommendations tailored for specific player segments within the True Survival Series. The focus is on enhancing player engagement through in-game farming mechanics and experiences.

### Bundle 1: Farm Automation Kit Expansion

#### Bundle Price
- **$9.99**

#### Bundle Items
- 5 Advanced Farming Tools (e.g., automated tillers)
- 10 Exclusive Seeds (with special attributes)
- 1 Farm Layout Template (easy to implement)

#### Bundle Description
Boost your automated farming with the **Farm Automation Kit Expansion**! This bundle includes advanced tools designed to enhance productivity, exclusive seeds that yield higher resources, and a blueprint for an efficient farm layout. Perfect for players who wish to maximize their farming potential without requiring design changes or additional development.

#### Segments
- Players engaged primarily in farming activities.
- Players with a high session frequency focused on sustaining larger farms.

#### Bundle Recommendation
Given the current player metrics, this bundle is recommended for those actively participating in farming, as seen by their session lengths and frequency. Players leveraging automated farming are likely to find value in this offer, potentially increasing their session time and overall engagement.

#### Product Experiment Instructions
1. Target the identified segment with in-game notifications showcasing the **Farm Automation Kit Expansion**.
2. Track engagement metrics (e.g., session time, session frequency) pre- and post-launch of the bundle offer.
3. Conduct surveys among players who purchase the bundle to gather feedback and insights on their farming experience.

### Conclusion
The **Farm Automation Kit Expansion** not only aligns with the gameplay mechanics but also addresses player interests in farming, thereby enhancing their overall experience in the True Survival Series. Monitoring the success of this offer will provide vital data for future personalised in-app purchase strategies.
